### PR TITLE
Streaming download

### LINKED ISSUE #13688: Simplify downloading clip
**Describe what you are trying to accomplish and why in non technical terms**
It takes multiple steps of navigation to download a clip to the local device.

**Describe the solution you'd like**
Add a "Download" button next to the "Export" button so a user can choose to download or export.

### PR DESCRIPTION
**Pull request by NickM-27**
## Proposed change

Implements the download API as a streaming response instead of a file response which is much more efficient and starts immediately vs waiting for the file to build

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code

## Additional information

- This PR fixes or closes https://github.com/blakeblackshear/frigate/issues/13688
- This PR is related to issue: 

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

File: web/src/components/overlay/detail/ReviewDetailDialog.tsx [hunk 1]
```diff
@@ -161,6 +161,19 @@ export default function ReviewDetailDialog({
...
+                    <a
+                      download
+                      href={`${baseUrl}api/${review.camera}/start/${review.start_time}/end/${review.end_time || Date.now() / 1000}/clip.mp4`}
+                    >
+                      <Button size="sm">
+                        <FaDownload className="size-4 text-secondary-foreground" />
+                      </Button>
+                    </a>
+                  </TooltipTrigger>
+                  <TooltipContent>Share this review item</TooltipContent>
```
Comment (hawkeye217) [inline web/src/components/overlay/detail/ReviewDetailDialog.tsx:175]:
```suggestion
                  <TooltipContent>Download</TooltipContent>
```

File: web/src/components/overlay/detail/ReviewDetailDialog.tsx [hunk 2]
```diff
@@ -143,7 +143,7 @@ export default function ReviewDetailDialog({
               <Description className="sr-only">Review item details</Description>
               <div
                 className={cn(
-                  "absolute",
+                  "absolute flex gap-2",
```
Comment (hawkeye217) [inline web/src/components/overlay/detail/ReviewDetailDialog.tsx:146]:
```suggestion
                  "absolute flex lg:flex-col gap-2",
```

**Review by hawkeye217 — APPROVED** [review]

File: web/src/components/button/DownloadVideoButton.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const formattedDate = formatUnixTimestampToDateTime(startTime, {
+      strftime_fmt: "%D-%T",
+      time_style: "medium",
+      date_style: "medium",
+    });
+    const filename = `${camera}_${formattedDate}.mp4`;
+
+    try {
+      const response = await fetch(source);
+      const blob = await response.blob();
```
Comment (leccelecce) [inline web/src/components/button/DownloadVideoButton.tsx:32]: @NickM-27 is this essentially reading the whole MP4 file from the server into the browser's memory, then creating a dummy link to click it to trigger the "download"? 

If so, that feels like a bit of a risk e.g. for low memory phones. 
https://chromium.googlesource.com/chromium/src/+/master/storage/browser/blob/README.md#blob-storage-limits also seems to suggest that on Android, Chrome might page blobs to disk at 20MB. A 4K, 25fps camera at a bitrate of 16mbps is ~ 60MB for a 30 second file.

Comment (hawkeye217) [reply web/src/components/button/DownloadVideoButton.tsx:32]: A direct download link is probably preferable. I'll put up another PR later.

Comment (hawkeye217) [reply web/src/components/button/DownloadVideoButton.tsx:32]: https://github.com/blakeblackshear/frigate/pull/14347

**Pull request merged**